src/main/java/com/exasol/bucketfs/ReadOnlyBucket.java

```diff
@@ -137,4 +152,13 @@ default String getPathInUdf() {
     default String getPathInUdf(final String fileInBucketFs) {
         return getPathInUdf() + PATH_SEPARATOR + fileInBucketFs;
     }
+
+    /**
+     * Get the HTTP client the bucket uses to issue its commands.
+     * <p>
+     * This is necessary for better observability and testability.
+     * </p>
+     * @return HTTP client the bucket uses.
+     */
+    HttpClient getHttpClient();
```

Comment:
Feels strange to expose the http client etc. Can you explain why this is needed?